zipline/data/data_portal.py

```diff
@@ -696,6 +767,26 @@ def get_history_window(self, assets, end_dt, bar_count, frequency, field,
 
         # forward-fill if needed
         if field == "price" and ffill:
+            assets_with_nan_index = np.where(pd.isnull(df.iloc[0, :]))[0]
+            assets_with_leading_nan = assets[assets_with_nan_index]
+
+            if frequency == "1m":
+                data_frequency = 'minute'
+            elif frequency == "1d":
+                data_frequency = 'daily'
+            else:
+                raise Exception(
+                    "Only 1d and 1m are supported for forward-filling.")
+
+            for assets in assets_with_leading_nan:
```

Comment:
I did start out with that model, but both getting answer of 'get the number of bars between beginning of window and the known last dt' a bit awkward to handle for both daily and minute (minute being the more awkward of the two.)

Also, went down the route of trying to use `get_minute_window_for_asset` with creating a a minute creating a minute window using `env.minute_market_window` but that method also expected the bar count.

We could add an analogue to `trading_day_distance` for `trading_minute_distance` using the `minutes_for_days_in_range`.

I also tried using a "fake" window of `DatetimeIndex([previous_dt, perspective_dt]), but ran into issues (which I can try to recreate) when passing that window `get_market_window_for_asset`.

I do concede that adding a different paths for adjustments is added complexity, but was trading it for the complexity in constructing window to pass to the existing adjustment code path.